sdk/keyvault/azure-security-keyvault-administration/README.md

```diff
@@ -130,6 +130,137 @@ To update the value of any of the the available settings, we will call the Updat
+{
+  std::this_thread::sleep_for(5s);
+  backupStatus = client.FullBackupStatus(backupStatus.Value.JobId);
+}
+std::cout << "Backup Job Id: " << backupStatus.Value.JobId << std::endl
+          << "Backup Status: " << backupStatus.Value.Status << std::endl;
+```
+
+## The FullRestore operation
+
+Similar to the backup operation after we initialize the operation we can check the status. 
+
+### Starting the restore operation 
+
+the restore operation requires a folder where a backup was previously performed along side the SAS token parameter. 
+```cpp Snippet:SampleAdministration2FullRestoreStart
+// Restore the full backup using a user-provided SAS token to an Azure blob storage container.
+RestoreOperationParameters restoreBlobDetails;
+restoreBlobDetails.SasTokenParameters = sasTokenParameter;
+restoreBlobDetails.FolderToRestore = "folder name here";
+std::cout << "Folder to restore: " << restoreBlobDetails.FolderToRestore << std::endl;
+auto restoreResponse = client.FullRestore(restoreBlobDetails);
+std::cout << "Restore Job Id: " << restoreResponse.Value.JobId << std::endl
+          << "Restore Status: " << restoreResponse.Value.Status << std::endl;
+```
+
+### FullRestore status check
+
+```cpp Snippet:SampleAdministration2StatusRestore
+// Get the status of the restore operation.
+auto restoreStatus = client.RestoreStatus(restoreResponse.Value.JobId);
+// Wait for the restore to complete.
+while (restoreStatus.Value.Status == "InProgress")
+{
+  std::this_thread::sleep_for(5s);
+  restoreStatus = client.RestoreStatus(restoreStatus.Value.JobId);
+}
+std::cout << "Restore Job Id: " << restoreStatus.Value.JobId << std::endl
+          << "Restore Status: " << restoreStatus.Value.Status << std::endl;
+```
+
+## The SelectiveRestore operation
+
+Similar to the backup operation after we initialize the operation we can check the status. 
+
+### Starting the restore operation 
+
+The selective restore operation requires a folder where a backup was previously performed along side the SAS token parameter. 
+
+```cpp Snippet:SampleAdministration2FullRestoreStart
+// Restore the full backup using a user-provided SAS token to an Azure blob storage container.
+RestoreOperationParameters restoreBlobDetails;
+restoreBlobDetails.SasTokenParameters = sasTokenParameter;
+restoreBlobDetails.FolderToRestore = "folder name here";
+std::cout << "Folder to restore: " << restoreBlobDetails.FolderToRestore << std::endl;
+auto selectiveRestore = client.SelectiveKeyRestore(keyName, restoreBlobDetails);
+std::cout << "Restore Job Id: " << restoreResponse.Value.JobId << std::endl
+          << "Restore Status: " << restoreResponse.Value.Status << std::endl;
+```
+
+### FullRestore status check
+
+```cpp Snippet:SampleAdministration2StatusRestore
+// Get the status of the restore operation.
+auto restoreStatus = client.RestoreStatus(restoreResponse.Value.JobId);
+// Wait for the restore to complete.
+while (restoreStatus.Value.Status == "InProgress")
```

Comment:
Shouldn't the `Status` be an extendable enumeration? And thus you would use `RestoreStatus::InProgress` rather than requiring that the customer hard code the string "InProgress" into their code (which is insanely fragile - lots of opportunities for typos).

Reply:
.Net relies on strings also , i used them as a template

Reply:
according to the .net guys since the enum is strings and it all bois down to string comparison at the end of the day they did not think it was any issue with not declaring it as an enum